aries_cloudagent/multitenant/askar_profile_manager.py

```diff
@@ -0,0 +1,91 @@
+"""Manager for askar profile multitenancy mode."""
+
+import logging
+
+from ..core.profile import (
+    Profile,
+)
+from ..config.wallet import wallet_config
+from ..config.injection_context import InjectionContext
+from ..wallet.models.wallet_record import WalletRecord
+from ..askar.profile import AskarProfile
+from ..multitenant.base import BaseMultitenantManager
+
+LOGGER = logging.getLogger(__name__)
+
+
+class AskarProfileMultitenantManager(BaseMultitenantManager):
+    """Class for handling askar profile multitenancy."""
+
+    DEFAULT_MULTIENANT_WALLET_NAME = "multitenant_sub_wallet"
+
+    def __init__(self, profile: Profile):
+        """Initialize askar profile multitenant Manager.
+
+        Args:
+            profile: The base profile for this manager
+        """
+        super().__init__(profile)
+
+    async def get_wallet_profile(
+        self,
+        base_context: InjectionContext,
+        wallet_record: WalletRecord,
+        extra_settings: dict = {},
+        *,
+        provision=False,
+    ) -> Profile:
+        """Get Askar profile for a wallet record.
+
+        Args:
+            base_context: Base context to extend from
+            wallet_record: Wallet record to get the context for
+            extra_settings: Any extra context settings
+
+        Returns:
+            Profile: Profile for the wallet record
+
+        """
+        multitenant_wallet_name = (
+            base_context.settings.get("multitenant.wallet_name")
+            or self.DEFAULT_MULTIENANT_WALLET_NAME
+        )
+
+        if multitenant_wallet_name not in self._instances:
+            context = base_context.copy()
+            # Settings we don't want to use from base wallet
+            reset_settings = {
```

Comment:
Maybe rename to update_settings & change comment, since this is not reset anymore but a change of settings.

Reply:
renamed to sub_wallet_settings I think this makes more sense